src/uct/tcp/tcp_ep.c

```diff
@@ -505,6 +563,48 @@ static inline unsigned uct_tcp_ep_send(uct_tcp_ep_t *ep)
     return send_length > 0;
 }
 
+static inline unsigned uct_tcp_ep_sendv(uct_tcp_ep_t *ep)
+{
+    uct_tcp_iface_t *iface      = ucs_derived_of(ep->super.super.iface,
+                                                 uct_tcp_iface_t);
+    uct_tcp_ep_zcopy_ctx_t *ctx = (uct_tcp_ep_zcopy_ctx_t*)ep->tx.buf;
+    struct iovec *ctx_iov       = (struct iovec*)ctx->iov;
+    size_t send_length;
+    ucs_status_t status;
+
+    ucs_assertv(ep->tx.offset < ep->tx.length, "ep=%p", ep);
+
+    /* Adjust the current IOV buffer to send from right point */
+    ctx_iov[ctx->iov_index].iov_len  -= ctx->iov_offset;
```

Comment:
maybe we shouldn't keep 'iov_offset' at all, and 'consume' function should advance ptr and reduce iov_len?